bidscoin/bids.py

```diff
@@ -1512,7 +1513,10 @@ def append_run(bidsmap: dict, run: dict, clean: bool=True) -> None:
             run_[item].update(run[item])
 
         run = run_
+    else:
+        run = copy.copy(run)  # popping targets will not change original run
 
```

Comment:
This looks a bit like a hack, I suspect it is not needed if we add `targets` to the datasource instead of to the run. The datasource as part of the run is already hacky, but at least then it is all in one place. Moreover, cleaning up the datasource from the run is already implemented (e.g. when saving the bidsmap). I will make some commits to your PR so you can see what I mean